rust/src/quic/frames.rs

```diff
@@ -210,6 +444,28 @@ fn parse_stream_frame(input: &[u8], frame_ty: u8) -> IResult<&[u8], Frame, QuicE
     ))
 }
 
+fn parse_crypto_stream_frame(input: &[u8]) -> IResult<&[u8], Frame, QuicError> {
+    let (rest, _offset) = quic_var_uint(input)?;
+    let (rest, data_length) = quic_var_uint(rest)?;
+    let (rest, stream_data) = take(data_length)(rest)?;
```

Comment:
On my ARM 32 bit builder I get an error here:
```
   Compiling suricata v7.0.0-dev (/builds/inliniac/suricata-ci/suricata/rust)
error[E0277]: the trait bound `u64: nom::ToUsize` is not satisfied
   --> src/quic/frames.rs:450:36
    |
450 |     let (rest, stream_data) = take(data_length)(rest)?;
    |                               ---- ^^^^^^^^^^^ the trait `nom::ToUsize` is not implemented for `u64`
    |                               |
    |                               required by a bound introduced by this call
    |
    = help: the following other types implement trait `nom::ToUsize`:
              u16
              u32
              u8
              usize
note: required by a bound in `nom::bytes::complete::take`
   --> /builds/inliniac/suricata-ci/suricata/rust/vendor/nom-5.1.2/src/bytes/complete.rs:381:6
    |
381 |   C: ToUsize,
    |      ^^^^^^^ required by this bound in `nom::bytes::complete::take`
```
`usize` is 32 bit on 32 bit platforms, so the question here is what to do about the case where `data_length` exceeds the max 32 value.

Reply:
As an aside: I'm surprised to see nom 5 in this error. I would expect new code to be all nom 7.

Reply:
> As an aside: I'm surprised to see nom 5 in this error. I would expect new code to be all nom 7.

I think this is not an aside, this is the main problem, or the root cause of it...

Reply:
@ct0br0 would we have a 32 bit build in Github CI ?

Reply:
@victorjulien next version should fix these